modules/xnft/src/xcm_helpers.rs

```diff
@@ -0,0 +1,97 @@
+// This file is part of Acala.
+
+// Copyright (C) 2023 Unique Network.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use crate::*;
+use xcm::v3::AssetId::Concrete;
+use xcm_executor::traits::Error as MatchError;
+
+impl<T: Config> Pallet<T>
+where
+	TokenIdOf<T>: TryFrom<u128>,
+	ClassIdOf<T>: TryFrom<u128>,
+{
+	pub fn asset_to_collection(asset: &AssetId) -> Result<(ClassIdOf<T>, bool), MatchError> {
```

Comment:
what is the bool in result? it is not obvious and either need to introduce a struct or very minimal lots of comments to explain it

Reply:
`ClassLocality` enum added. Refactored.